File: clients/java/Tile.java
import com.sun.jna.Pointer;

///Represents a single tile on the map.
class Tile extends Mappable
{
  public Tile(Pointer p)
  {
    super(p);
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.tiles.length; i++)
    {
      if(BaseAI.tiles[i].ID == ID)
      {
        ptr = BaseAI.tiles[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands


    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.tileGetId(ptr);
  }
  ///X position of the object
  public int getX()
  {
    validify();
    return Client.INSTANCE.tileGetX(ptr);
  }
  ///Y position of the object
  public int getY()
  {
    validify();
    return Client.INSTANCE.tileGetY(ptr);
  }
  ///Owner of spawning droid. 0 - Player 1, 1 - Player 2, 2 - No spawning droid.
  public int getOwner()
  {
    validify();
    return Client.INSTANCE.tileGetOwner(ptr);
  }
  ///The number of turns until a Droid is assembled.
  public int getTurnsUntilAssembled()
  {
    validify();
    return Client.INSTANCE.tileGetTurnsUntilAssembled(ptr);
  }
  ///The variant of Droid to assemble.
  public int getVariantToAssemble()
  {
    validify();
    return Client.INSTANCE.tileGetVariantToAssemble(ptr);
  }

}

File: clients/csharp/Tile.cs
using System;
using System.Runtime.InteropServices;

/// <summary>
/// Represents a single tile on the map.
/// </summary>
public class Tile: Mappable
{

  public Tile()
  {
  }

  public Tile(IntPtr p)
  {
    ptr = p;
    ID = Client.tileGetId(ptr);
    iteration = BaseAI.iteration;
  }

  public override bool validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.tiles.Length; i++)
    {
      if(BaseAI.tiles[i].ID == ID)
      {
        ptr = BaseAI.tiles[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

  #region Commands
  #endregion

  #region Getters
  /// <summary>
  /// Unique Identifier
  /// </summary>
  public new int Id
  {
    get
    {
      validify();
      int value = Client.tileGetId(ptr);
      return value;
    }
  }

  /// <summary>
  /// X position of the object
  /// </summary>
  public new int X
  {
    get
    {
      validify();
      int value = Client.tileGetX(ptr);
      return value;
    }
  }

  /// <summary>
  /// Y position of the object
  /// </summary>
  public new int Y
  {
    get
    {
      validify();
      int value = Client.tileGetY(ptr);
      return value;
    }
  }

  /// <summary>
  /// Owner of spawning droid. 0 - Player 1, 1 - Player 2, 2 - No spawning droid.
  /// </summary>
  public int Owner
  {
    get
    {
      validify();
      int value = Client.tileGetOwner(ptr);
      return value;
    }
  }

  /// <summary>
  /// The number of turns until a Droid is assembled.
  /// </summary>
  public int TurnsUntilAssembled
  {
    get
    {
      validify();
      int value = Client.tileGetTurnsUntilAssembled(ptr);
      return value;
    }
  }

  /// <summary>
  /// The variant of Droid to assemble.
  /// </summary>
  public int VariantToAssemble
  {
    get
    {
      validify();
      int value = Client.tileGetVariantToAssemble(ptr);
      return value;
    }
  }

  #endregion

  #region Properties
  #endregion
}
